javav2/example_code/firehose/src/main/java/com/example/firehose/scenario/FirehoseScenario.java

```diff
@@ -0,0 +1,232 @@
+
+package com.example.firehose.scenario;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchClient;
+import software.amazon.awssdk.services.cloudwatch.model.*;
+import software.amazon.awssdk.services.firehose.FirehoseClient;
+import software.amazon.awssdk.services.firehose.model.PutRecordBatchRequest;
+import software.amazon.awssdk.services.firehose.model.PutRecordBatchResponse;
+import software.amazon.awssdk.services.firehose.model.PutRecordRequest;
+import software.amazon.awssdk.services.firehose.model.Record;
+import software.amazon.awssdk.regions.Region;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Before running this Java V2 code example, set up your development
+ * environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ */
+
+// snippet-start:[firehose.java2.scenario.main]
+public class FirehoseScenario {
+    private static FirehoseClient firehoseClient;
+    private static CloudWatchClient cloudWatchClient;
+    private static String deliveryStreamName;
+
+    public static void main(String[] args) {
+        firehoseClient = FirehoseClient.builder().region(Region.US_EAST_1).build();
+        cloudWatchClient = CloudWatchClient.builder().region(Region.US_EAST_1).build();
+
+        // Replace with your region and delivery stream name
+        deliveryStreamName = "stream35";
+
+        /*
+           See the Readme in the scenario folder for information about the sample_records.json file.
+           After you create this file, place in your project's resources folder After you perform these tasks,
+           the sample data file can be used in this scenario.
+         */
+        String jsonContent = readJsonFile("sample_records.json");
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            List<Map<String, Object>> sampleData = objectMapper.readValue(
+                jsonContent,
+                new TypeReference<>() {}
+            );
+
+            // Process individual records
+            System.out.println("Processing individual records...");
+            sampleData.subList(0, 100).forEach(record -> {
+                try {
+                    putRecord(record);
+                } catch (Exception e) {
+                    System.out.println("Put record failed: " + e.getMessage());
+                }
+            });
+
+            monitorMetrics();
+
+            // Process batch records.
+            System.out.println("Processing batch records...");
+            putRecordBatch(sampleData.subList(100, 200), 50);
```

Comment:
This should not be a sublist of the next 100 records.  This only uses a small subset of the sample records. This should be a sublist of all the records except the first 100. Also, the python example uses a default batch size of 500.
```suggestion
            putRecordBatch(sampleData.subList(100, sampleData.size()), 50);
```

Reply:
fixed